R/import_methods.R

```diff
@@ -218,24 +218,17 @@ function(file,
 #' @importFrom haven read_dta
 #' @export
 .import.rio_dta <- function(file, haven = TRUE,
-                            convert.dates = TRUE,
-                            convert.factors = FALSE,
-                            missing.type = FALSE, ...) {
-    if (isTRUE(haven)) {
-        a <- list(...)
-        if (length(a)) {
-            warning("File imported using haven. Arguments to '...' ignored.")
-        }
-        standardize_attributes(haven::read_dta(file = file))
-    } else {
-        out <- foreign::read.dta(file = file,
-                    convert.dates = convert.dates,
-                    convert.factors = convert.factors,
-                    missing.type = missing.type, ...)
-        attr(out, "expansion.fields") <- NULL
-        attr(out, "time.stamp") <- NULL
-        standardize_attributes(out)
-    }
+                               convert.factors = FALSE,...) {
+  if (isTRUE(haven)) {
+    arg_reconcile(haven::read_dta, file = file, ..., .docall = TRUE,
+                  .finish = standardize_attributes)
```

Comment:
The `.finish` argument of `arg_reconcile()` is a function to run on the output before returning it. Since in this case, the only post-processing step is `standardize_attributes()`, it can be passed via the `.finish` argument. Below, for the `read.dta()` case, there are multiple steps so those still need to be done outside of `arg_reconcile()`.